### ISSUE TITLE
ENH: Add Alpine to CI

### ISSUE THREAD

**Issue by WillAyd** [pandas-dev/pandas#50511]
### Feature Type

- [X] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [X] Removing existing functionality in pandas

### Problem Description

We have historically had some bug reports come from Alpine users but have never tested that platform in the past. Now that we have an open source Docker account we should be able to create an Alpine image for testing use

Related to #50046

### Feature Description

see description

### Alternative Solutions

I don't believe GH actions has an alpine image, but if it does that could work too

**Comment by lithomas1** [comment]
I don't think we should commit to this until numpy tests Alpine in their CI.

IIRC, we do have a good chunk of users on Alpine. I think I ran the stats a couple of months back, and they were one of the most common platforms installing from sdist.

**Comment by WillAyd** [comment]
Yea my guess is Alpine will be the container of choice for any deployments of pandas within lambda functions

**Comment by kostyafarber** [comment]
Should we open an issue in nummy asking them to add Alpine to their CI?

**Comment by WillAyd** [comment]
Sure that would be awesome @kostyafarber

**Comment by kostyafarber** [comment]
@WillAyd numpy have closed the issue so I guess we can start making tracks on this now

**Comment by WillAyd** [comment]
Awesome!

**Comment by kostyafarber** [comment]
Happy to assist on this if needed. Pointing towards any resources/instructions would be helpful.

**Comment by WillAyd** [comment]
Awesome! Yea the pandas CI is housed in the .github folder - you might be particularly interested in the Workflows therein to extend or model this after. 

We need a source image to start from. I don't know if GitHub actions has a musl image to build off of. If not I would see what numpy is using as their foundation, or could also use the alpine image from https://github.com/pandas-dev/pandas/pull/50046

**Comment by sanmai-NL** [comment]
@WillAyd You can just use the Alpine Linux image, which is the main Linux distribution using musl libc right now.

**Comment by lithomas1** [comment]
Closing as this is completed.

**Comment by MrMino** [comment]
@lithomas1 @WillAyd do you have any plans for providing `musllinux` wheels (for PyPI)?

**Comment by WillAyd** [comment]
I think would be open to it but needs a community PR - would you be interested in trying that @MrMino ?

Note that we just recently approved PDEP-10 https://github.com/pandas-dev/pandas/pull/52711 so we probably need pyarrow to distribute a musllinux wheel first.

**Comment by MrMino** [comment]
Yup, seems like the relevant [CI stuff already there](https://github.com/pandas-dev/pandas/blob/main/.github/workflows/wheels.yml#L92C26-L92C42).

I also nudged the issue on the `pyarrow` side, it has been discussed before PEP-656 happened.

**Comment by MrMino** [comment]
FTR `pyarrow` released musllinux wheels in v20.0.0

See https://github.com/apache/arrow/issues/18036
